### PR TITLE
Improved documentation

### PR DESCRIPTION
**Pull request by abhinavsagar**
Added documentation in `q_learning_impl` and `async_learning_impl`

### CODE REVIEW

File: src/mlpack/methods/reinforcement_learning/async_learning_impl.hpp [hunk 1]
```diff
@@ -19,6 +19,13 @@
 namespace mlpack {
 namespace rl {
 
+ /**
+   * @tparam WorkerType The type of the worker.
```
Comment (zoq) [inline src/mlpack/methods/reinforcement_learning/async_learning_impl.hpp]: This is somewhat redundant this it's already part of the header file.

File: src/mlpack/methods/reinforcement_learning/q_learning_impl.hpp [hunk 2]
```diff
@@ -17,6 +17,13 @@
 namespace mlpack {
 namespace rl {
 
+  /**
+    * @tparam EnvironmentType The type of the reinforcement learning task.
```
Comment (zoq) [inline src/mlpack/methods/reinforcement_learning/q_learning_impl.hpp]: See comment above.

File: src/mlpack/methods/reinforcement_learning/q_learning_impl.hpp [hunk 3]
```diff
@@ -67,6 +75,7 @@ arma::Col<size_t> QLearning<
   ReplayType
 >::BestAction(const arma::mat& actionValues)
 {
+  // Take best possible action at a particular instance
```
Comment (zoq) [inline src/mlpack/methods/reinforcement_learning/q_learning_impl.hpp]: Missing point at the end.

File: src/mlpack/methods/reinforcement_learning/q_learning_impl.hpp [hunk 4]
```diff
@@ -45,6 +52,7 @@ QLearning<
     totalSteps(0),
     deterministic(false)
 {
+  // Set up q learning network.
```
Comment (zoq) [inline src/mlpack/methods/reinforcement_learning/q_learning_impl.hpp]: Should we go with `q-learning`?

Comment (ShikharJ) [reply src/mlpack/methods/reinforcement_learning/q_learning_impl.hpp]: I think that is reasonable.

**Comment by zoq** [comment]
Thanks again for the contribution.

**Pull request merged**